File: app/src/androidTest/java/testhelper/FragmentTestRule.java
package testhelper;

import android.support.test.rule.ActivityTestRule;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;

import junit.framework.Assert;

import eu.darken.mvpbakery.injection.ManualInjector;
import eu.thedarken.wldonate.FragmentTestActivity;

public class FragmentTestRule<FragmentT extends Fragment> extends ActivityTestRule<FragmentTestActivity> {

    private final Class<FragmentT> fragmentClass;
    private FragmentT fragment;
    ManualInjector<Fragment> manualInjector;


    public FragmentTestRule(final Class<FragmentT> fragmentClass) {
        super(FragmentTestActivity.class, true, false);
        this.fragmentClass = fragmentClass;
    }

    @Override
    protected void afterActivityLaunched() {
        super.afterActivityLaunched();

        getActivity().setManualInjector(manualInjector);

        getActivity().runOnUiThread(() -> {
            try {
                //Instantiate and insert the fragment into the container layout
                FragmentManager manager = getActivity().getSupportFragmentManager();
                FragmentTransaction transaction = manager.beginTransaction();
                fragment = fragmentClass.newInstance();
                //noinspection ResourceType
                transaction.replace(1, fragment);
                transaction.commitNow();
            } catch (InstantiationException | IllegalAccessException e) {
                Assert.fail(
                        String.format("%s: Could not insert %s into TestActivity: %s",
                                getClass().getSimpleName(),
                                fragmentClass.getSimpleName(),
                                e.getMessage())
                );
            }
        });
    }

    public FragmentT getFragment() {
        return fragment;
    }

    public void setManualInjector(ManualInjector<Fragment> manualInjector) {
        this.manualInjector = manualInjector;
    }
}

File: app/src/androidTest/java/testhelper/FragmentTestRule.kt
package testhelper

import androidx.test.rule.ActivityTestRule
import eu.darken.mvpbakery.injection.ManualInjector
import eu.thedarken.wldonate.FragmentTestActivity
import junit.framework.Assert

class FragmentTestRule<FragmentT : androidx.fragment.app.Fragment>(private val fragmentClass: Class<FragmentT>) : ActivityTestRule<FragmentTestActivity>(FragmentTestActivity::class.java, true, false) {
    var fragment: FragmentT? = null
        private set
    internal lateinit var manualInjector: ManualInjector<androidx.fragment.app.Fragment>

    override fun afterActivityLaunched() {
        super.afterActivityLaunched()

        activity.setManualInjector(manualInjector)

        activity.runOnUiThread {
            try {
                //Instantiate and insert the fragment into the container layout
                val manager = activity.supportFragmentManager
                val transaction = manager.beginTransaction()
                fragment = fragmentClass.newInstance()

                transaction.replace(1, fragment!!)
                transaction.commitNow()
            } catch (e: InstantiationException) {
                Assert.fail(
                        String.format("%s: Could not insert %s into TestActivity: %s",
                                javaClass.simpleName,
                                fragmentClass.simpleName,
                                e.message)
                )
            } catch (e: IllegalAccessException) {
                Assert.fail(String.format("%s: Could not insert %s into TestActivity: %s", javaClass.simpleName, fragmentClass.simpleName, e.message))
            }
        }
    }

    fun setManualInjector(manualInjector: ManualInjector<androidx.fragment.app.Fragment>) {
        this.manualInjector = manualInjector
    }
}